software-delivery-performance-terraform/sumologic.tf

```diff
@@ -0,0 +1,242 @@
+resource "sumologic_collector" "sdp_collector" {
+  name     = var.collector_name
+  category = "SDP"
+}
+
+# Create/Delete Jira Cloud Source
+resource "sumologic_http_source" "jira_cloud" {
+  count        = "${var.install_jira_cloud}" ? 1 : 0
+  name         = "Jira Cloud"
+  category     = var.jira_cloud_sc
+  collector_id = sumologic_collector.sdp_collector.id
+}
+
+# Create/Delete BitBucket Cloud Source
+resource "sumologic_http_source" "bitbucket_cloud" {
+  count        = "${var.install_bitbucket_cloud}" ? 1 : 0
+  name         = "Bitbucket Cloud"
+  category     = var.bitbucket_sc
+  fields       = { "_convertHeadersToFields" = "true" }
+  collector_id = sumologic_collector.sdp_collector.id
+}
+
+# Create/Delete Jira Server Source
+resource "sumologic_http_source" "jira_server" {
+  count        = "${var.install_jira_server}" ? 1 : 0
+  name         = "Jira Server"
+  category     = var.jira_server_sc
+  collector_id = sumologic_collector.sdp_collector.id
+}
+
+# Create/Delete OpsGenie Source
+resource "sumologic_http_source" "opsgenie" {
+  count        = "${var.install_opsgenie}" ? 1 : 0
+  name         = "OpsGenie"
+  category     = var.opsgenie_sc
+  collector_id = sumologic_collector.sdp_collector.id
+}
+
+# Create/Delete Pagerduty Source
+resource "sumologic_http_source" "pagerduty" {
+  count        = "${var.install_pagerduty}" ? 1 : 0
+  name         = "Pagerduty"
+  category     = var.pagerduty_sc
+  collector_id = sumologic_collector.sdp_collector.id
+}
+
+# Create/Delete Github Source
+resource "sumologic_http_source" "github" {
+  count        = "${var.install_github}" ? 1 : 0
+  name         = "Github"
+  category     = var.github_sc
+  fields       = { "_convertHeadersToFields" = "true" }
+  collector_id = sumologic_collector.sdp_collector.id
+}
+
+# Create/Delete Jenkins Source
+resource "sumologic_http_source" "jenkins" {
+  count        = "${var.install_jenkins}" ? 1 : 0
+  name         = "Jenkins"
+  category     = var.jenkins_sc
+  fields       = { "_convertHeadersToFields" = "true" }
+  collector_id = sumologic_collector.sdp_collector.id
+}
+
+# Sumo Logic - Create/Delete folder for installing the applications
+data "sumologic_personal_folder" "personalFolder" {}
+
+resource "sumologic_folder" "folder" {
+  name        = var.app_installation_folder
+  description = "Atlassian Applications"
+  parent_id   = data.sumologic_personal_folder.personalFolder.id
+  depends_on  = [sumologic_collector.sdp_collector]
+}
+
+# Remove slash from Sumo Logic sumo_api_endpoint
+
+locals {
+  sumo_api_endpoint_fixed = replace(var.sumo_api_endpoint, "api/", "api")
+}
+
+# Install Apps
+# Install Jira Cloud
+resource "null_resource" "install_jira_cloud_app" {
+  count      = "${var.install_jira_cloud}" ? 1 : 0
+  depends_on = [sumologic_http_source.jira_cloud]
+
+  provisioner "local-exec" {
+    command = <<EOT
+        curl -s --request POST '${local.sumo_api_endpoint_fixed}/v1/apps/019757ca-3b08-457c-bd15-7239f1ab66c9/install' \
```

Comment:
Do we still need to use curl directly? Is there a terraform action for app installation now?

Reply:
Also for AWSO, we create a folder with timestamp suffix, and install all apps there.

Reply:
No Terraform resource for App installation. The recent one which was released was for Webhooks. https://www.terraform.io/docs/providers/sumologic/index.html

Reply:
> Also for AWSO, we create a folder with timestamp suffix, and install all apps there.

Good idea, will take a look.

Reply:
> > Also for AWSO, we create a folder with timestamp suffix, and install all apps there.
> 
> Good idea, will take a look.

Done